cloudmock/aws/mockautoscaling/group.go

```diff
@@ -18,52 +18,129 @@ package mockautoscaling
 
 import (
 	"fmt"
+	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/golang/glog"
 )
 
 func (m *MockAutoscaling) AttachInstances(input *autoscaling.AttachInstancesInput) (*autoscaling.AttachInstancesOutput, error) {
-	for _, group := range m.Groups {
-		if aws.StringValue(group.AutoScalingGroupName) == aws.StringValue(input.AutoScalingGroupName) {
-			for _, instanceID := range input.InstanceIds {
-				group.Instances = append(group.Instances, &autoscaling.Instance{InstanceId: instanceID})
-			}
-		}
+	m.mutex.Lock()
```

Comment:
Being picky but do we need func comments how this works for mocking. This is a lot of useful code, only if developers understand how to reuse it for testing.